app/src/main/java/com/example/drawingactivity/ColorUtils.kt

```diff
@@ -0,0 +1,94 @@
+package com.example.drawingactivity
+
+import android.graphics.Color
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+object ColorUtils {
+
+    private val HEXADECIMAL_PATTERN: Pattern = Pattern.compile("\\p{XDigit}+")
+
+    /**
+     * A data class that contains color values.
+     * Contains: Hue, Saturation, Value, Red, Green, Blue, and Hex Value.
+     */
+    data class ColorState(
+        var hue: Int,
+        var saturation: Float,
+        var value: Float,
+        var red: Int,
```

Comment:
why are we taking rgb values

Reply:
I use the dataclass to update all of my views. I feel like including the RGB here in this case is ok since this isn't being held outside of updating the views. My viewModel only holds HSV values and creates this from there